### PR TITLE
KNN implementation in R language

### LINKED ISSUE #1791: KNN Algorithm in R language
@adishjain @nj4710 @ayushin78
As a GSSoC participant, I would like to work on this, please assign this task to me.

### PR DESCRIPTION
**Pull request by JasbirCodeSpace**
Fixes #1791 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
Implementation of KNN in R language without any predefined functions

#### Languages Used:
R

#### Files Added:
KNN.R

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by raghav-dalmia** [comment]
@JasbirCodeSpace Remove extra empty lines and print statements.

**Comment by JasbirCodeSpace** [comment]
commit the updated code please review again

**Comment by JasbirCodeSpace** [comment]
Fixes #1791 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
Implementation of KNN in JS

#### Languages Used:
JS

#### Files Added:
KNN.html

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

File: KNN/KNN.R [hunk 1]
```diff
@@ -0,0 +1,51 @@
+file <- readline(prompt = "Enter file link : ")
```
Comment (jainaman224) [inline KNN/KNN.R]: Please right full name
K_Nearest_Neighbour as file and folder name.

File: KNN/KNN.R [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+train_size = floor(0.50 * nrow(Iris_Data))
+set.seed(1223)
+train_index = sample(1:nrow(Iris_Data), size = train_size)
+train_sample = Iris_Data[train_index, ]
+test_sample = Iris_Data[-train_index, ]
+train_label = train_sample[, 5]
+
+# distance calculation method
+distance <- function(train, test) {
+  dist = 0
```
Comment (jainaman224) [inline KNN/KNN.R]: Use 4 space indentation.

File: KNN/KNN.R [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    
+  }
+  table_data = table(labels)
+  new_label = names(table_data)[which(table_data == max(table_data))]
+  return(new_label)
+}
+
+test_data = test_sample[37, ]          # any random test sample
+new_label = KNN(test_data, 25)         # 25 nearest neighbours
+test_data[, 5] = new_label             # assign the predicted label
```
Comment (jainaman224) [inline KNN/KNN.R]: Host image to some place and add link as output.

File: KNN/KNN.html [hunk 4]
```diff
@@ -0,0 +1,146 @@
+<!DOCTYPE html>
```
Comment (jainaman224) [inline KNN/KNN.html]: Remove this file.

**Comment by jainaman224** [comment]
@JasbirCodeSpace Are you working on it?

**Comment by jainaman224** [comment]
@JasbirCodeSpace We have to close PR, if you remain unresponsive.

**Comment by JasbirCodeSpace** [comment]
Yes, I'm working on it

**Comment by MastersAbh** [comment]
Hi @JasbirCodeSpace I dont see any update from your side. Closing the PR for now, if you still want to work on this issue, reach out to us.

**Pull request closed without merging**